static/app/utils/withPlugins.tsx

```diff
@@ -15,76 +13,82 @@ type WithPluginProps = {
-          const prevProject = prevProps.project || (prevContext && prevContext.project);
-
-          // If previous org/project is undefined then it means:
-          // the HoC has mounted, `fetchPlugins` has been called (via cDM), and
-          // store was updated. We don't need to fetchPlugins again (or it will cause an infinite loop)
-          //
-          // This is for the unusual case where component is mounted and receives a new org/project prop
-          // e.g. when switching projects via breadcrumbs in settings.
-          if (!defined(prevProject) || !defined(prevOrg)) {
-            return;
-          }
-
-          const isOrgSame = prevOrg.slug === organization.slug;
-          const isProjectSame = prevProject.slug === project?.slug;
-
-          // Don't do anything if org and project are the same
-          if (isOrgSame && isProjectSame) {
-            return;
-          }
-
-          this.fetchPlugins();
-        },
-
-        fetchPlugins() {
-          const {organization, project} = this.props;
-
-          if (!project || !organization) {
-            return;
-          }
-
-          fetchPlugins({projectId: project.slug, orgId: organization.slug});
-        },
-
-        render() {
-          return (
-            <WrappedComponent
-              {...(this.props as P & WithPluginProps)}
-              plugins={this.state.store}
-            />
-          );
-        },
-      })
-    )
-  );
+) {
+  class WithPlugins extends React.Component<
+    Omit<P, keyof 'plugins'> & WithPluginProps,
+    State
+  > {
+    static displayName = `withPlugins(${getDisplayName(WrappedComponent)})`;
+    state = {plugins: [], loading: true};
+
+    componentDidMount() {
+      this.fetchPlugins();
+    }
+
+    componentDidUpdate(prevProps, _prevState, prevContext) {
+      const {organization, project} = this.props;
+
+      // Only fetch plugins when a org slug or project slug has changed
+      const prevOrg = prevProps.organization || (prevContext && prevContext.organization);
+      const prevProject = prevProps.project || (prevContext && prevContext.project);
+
+      // If previous org/project is undefined then it means:
+      // the HoC has mounted, `fetchPlugins` has been called (via cDM), and
+      // store was updated. We don't need to fetchPlugins again (or it will cause an infinite loop)
+      //
+      // This is for the unusual case where component is mounted and receives a new org/project prop
+      // e.g. when switching projects via breadcrumbs in settings.
+      if (!defined(prevProject) || !defined(prevOrg)) {
+        return;
+      }
+
+      const isOrgSame = prevOrg.slug === organization.slug;
+      const isProjectSame = prevProject.slug === project?.slug;
+
+      // Don't do anything if org and project are the same
+      if (isOrgSame && isProjectSame) {
+        return;
+      }
+
+      this.fetchPlugins();
+    }
+    componentWillUnmount() {
```

Comment:
nit but usually we have spaces around our methods I think?

Maybe there's a lint rule for this